vault/expiration.go

```diff
@@ -969,13 +980,24 @@ func (m *ExpirationManager) expireID(leaseID string) {
 			return
 		default:
 		}
+
+		m.coreStateLock.RLock()
+		if m.quitContext.Err() == context.Canceled {
```

Comment:
Small nit, but this might be a better way of handling this case:
```
select {
case <- m.quitContext.Done():
	m.logger.Error("expiration: core context canceled, not attempting further revocation of lease", "lease_id", leaseID)
	m.coreStateLock.RUnlock()
	returndefault:
}
```

Reply:
This should be safe; see https://golang.org/pkg/context/#pkg-variables

> Canceled is the error returned by Context.Err when the context is canceled.

It's defined behavior so should be reliable.